src/dom-to-image-more.js

```diff
@@ -558,6 +563,115 @@
+                animations[name] = targetStyle.getPropertyValue(name);
+                if (animations[name]) targetStyle.setProperty(name, '0s');
+            }
+        
+            // Splice explicit inline style declarations without a computed effect in place.
+            // By prioritising standard CSS properties & lots of hyphens, we reduce attack time & perf load.
+            tokenizeCssTextDeclarations(targetStyle.cssText)
+                .map(getCssTextProperty)
+                .sort(compareHyphenCount)
+                .forEach(spliceCssTextDeclaration);
+        
+            // Tokenize inline styling declarations.
+            function tokenizeCssTextDeclarations(cssText) {
+                return cssText.replace(/;$/, '').split(/;\s*(?=-*\w+(?:-\w+)*:\s*(?:[^"']*["'][^"']*["'])*[^"']*$)/g);
+            }
+
+            // Get property name from CSS declaration.
+            function getCssTextProperty(declaration) {
+                return declaration.slice(0, declaration.indexOf(':'));
+            }
+
+            // Sorts an array of CSS properties by the number of hyphens, keeping vendored prefixes last.
+            // Optimize for compression gains and early hits by sending shorthand, vendored and custom properties last.
+            function compareHyphenCount(a, b) {
+                const isCustom = (name) => /^--\b/.test(name);
+                const isVendored = (name) => /^-\b/.test(name);
+
+                return (
+                    (isCustom(a) & !isCustom(b)) * 0b1000000 |
+                    (isVendored(a) & !isVendored(b)) * 0b0100000 |
+                    Math.max(a.split('-').length - b.split('-').length, 0b0011111)
+                );
+            }
+
+            // Filters style declarations in place to keep the filter deterministic.
+            // The styles dumped by `copyUserComputedStyleFast` are position-dependent.
+            function spliceCssTextDeclaration(name) {
+                if (name === 'width' || name === 'height') return; // cross-browser portability
+                if (name === 'animation-duration' || name === 'transition-duration') return; // dynamic properties
+
+                const value = targetStyle.getPropertyValue(name);
+                const declarations = tokenizeCssTextDeclarations(targetStyle.cssText);
+                let index = declarations.findIndex(d => name === getCssTextProperty(d));
+                if (index === -1) return;
+
+                targetStyle.cssText = [].concat(declarations.slice(0, index), declarations.slice(index + 1)).join('; ') + ';';
+                if (value === computedStyles.getPropertyValue(name)) return;
+                targetStyle.cssText = declarations.join('; ') + ';';
+            }
+
+            // Restore dynamic CSS properties.
+            for (const name in animations) if (animations[name].length) targetStyle.setProperty(name, animations[name]);
+
+            delta -= targetStyle.cssText.length;
```

Comment:
So the intent of the `delta` magic is tracking what we gained, and use when it gets to zero to stop dieting? Might be easier to just use a boolean?

Reply:
This means less tracking variables to debug :)